simavr/sim/avr_timer.c

```diff
@@ -462,7 +462,7 @@ avr_timer_write(
 			// to be synced. To obtain better simulation results
 			// p->tov_base type must be float or avr->frequency
 			// must be multiple of 32768.
-			p->cs_div_clock = avr->frequency / 32768;
+			p->cs_div_clock = (uint32_t)((float)avr->frequency * (1 << p->cs_div[new_cs]) / 32768);
```

Comment:
That's the only bit I don't like here; floats are 'random' in my book. Are you sure casting to uint64_t isn't enough? Or if you want to sacrifice a little precision, do it the other way around: ((frequency / 32768) * (1 << ...)) or some equivalent..